web-server/v0.4/e2etests

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+cd $(dirname $0)
+cat > mock/datastoreConfig.js <<EOF
+export default {
+  '/dev/datastoreConfig': {
+    elasticsearch: 'http://test_domain.com',
+    results: 'http://test_domain.com',
+    graphql: 'http://test_domain.com',
+    prefix: 'test_prefix.',
+    run_index: 'test_index.',
+  },
+};
+EOF
+yarn --network-timeout 1000000
+if [[ $? -ne 0 ]]; then
+  echo "yarn failed!" >&2
+  exit 1
+fi
+yarn start &
+if [[ $? -ne 0 ]]; then
+  echo "yarn failed!" >&2
+  exit 1
+fi
+sleep 60
```

Comment:
Why do we need to sleep for 60 seconds?  Is there another way we can tell with yarn start is ready?  Does it emit some kind of output that we can check for?

Can you add a comment describing the requirement?

Let's not change this now, except for adding a comment, but perhaps we can consider doing the following, which seems like a lot to avoid a 60 second sleep, but it can help reduce the amount of time these e2e jobs run:

```
yarn start > yarn.stdout 2> yarn.stderr < /dev/null &
yarn_pid=${!}
counter=0
max_tries=60
started=""
while [[ ${started} == "" && ${counter} < ${max_tries} ]]; do
    if [[ -n ${yarn_stderr} ]]; then
        wait ${yarn_pid}
        echo "Errors encountered from yarn"
        cat ${yarn_stderr}
        exit 1
    fi
    started=$(grep -F "started" yarn.stdout)
    if [[ ${counter} != 0 ]]; then
        # Don't sleep on the first iteration
        sleep 1
    fi
    let counter=counter+1
done
if [[ ${counter} >= ${max_tries} ]]; then
    wait ${yarn_pid}
    echo "Errors encountered from yarn"
    cat ${yarn_stderr}
    exit 1
fi
```

I'll propose a follow-up PR with a change here.

Reply:
In generarl, `yarn start` takes an x amount of time for compiling the dashboard. Another way we can tell the dashboard has been built successfully within a local environment is verifying if the `yarn start` command emits "Compiled successfully". Adding this dynamic check over a sleep would definitely speed up the build tests.